### PR TITLE
added verification of info about the current user

### PR DESCRIPTION
**Pull request by halyna-k**

### CODE REVIEW

File: src/components/Authentication/Authentication.js [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
+    user,
+    getAccessTokenSilently,
+    loginWithRedirect,
+    logout,
+  } = useAuth0();
+
+  const [userMetaData, setUserMetaData] = useState(null);
+
+  useEffect(() => {
+      const getUserMetaData = async () => {
```
Comment (dimalian) [inline src/components/Authentication/Authentication.js:22]: The data for the user profile must be obtained from the back-end. Profile data will not always be the same as the data from auth0.

File: src/components/Authentication/Authentication.js [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+        const domain = AUTH0_USER_DOMAIN;
+        try {
+          const accessToken = await getAccessTokenSilently({
+            authorizationParams: {
+               audience: `https://${domain}/api/v2/`,
+               scope: 'read:current_user',
+            },
+           });
```
Comment (dimalian) [inline src/components/Authentication/Authentication.js:30]: You need to get this token and put it to the Axios config.
You can do it in this way `axiosClient.interceptors.request.use((config) => {
        config.headers.Authorization = `Bearer ${token}``

**Comment by dimalian** [comment]
You need to make a small refactor because your authorization structure is wrong.
1. You need to rename `AuthProvider` to `Auth0Provider` and wrap it around not only the `HeaderButton` but the whole app like this:
`<Auth0Privder> <App/></Auth0Provider>`
2. You need to create `AuthProvider.js `and get the token as you did in the file `Authentication.js` and put it to the Axios client like this:
  ```
    useEffect(() => {
        authData
          .getAccessTokenSilently({
            authorizationParams: {
              audience, // Value in Identifier field for the API being called.
            },
          })
          .then(token => axiosClient.interceptors.request.use((config) => {
            config.headers.Authorization = `Bearer ${token}`;
          }))
          .catch(error => console.log(error));
      }, [audience, authData])
```
and wrap the `App` with this provider: 

```
<Auth0Privder>
  <AuthPrivder>
     <App/>
  </AuthPrivder>
</Auth0Provider>
```
4. Put all these providers in the separate folder `providers` inside the `src` folder

**Review by dimalian — CHANGES_REQUESTED** [review]

File: src/providers/AuthProvider.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  useEffect(() => {
+     getAccessTokenSilently({
+       authorizationParams: {
+         audience: AUTH0_USER_AUDIENCE,
+       },
+     })
+       .then((token) =>
+         axiosClient.interceptors.request.use((config) => {
+           config.headers.Authorization = `Bearer ${token}`;
+           console.log(token);
```
Comment (dimalian) [inline src/providers/AuthProvider.js]: Remove it

File: .vscode/settings.json
Comment (dimalian) [inline .vscode/settings.json:1]: Add the folder .vscode to the gitignore. Not everyone uses vscode

**Review by dimalian — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (halyna-k) [reply .vscode/settings.json:1]: Okay

**Pull request merged**